docs/blog/2021-07-18-macro.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Complex Procedural Rust Macros
+author: Andre Bogus
+author_title: Chief Rustacean
+author_url: https://github.com/getsynth
+author_image_url: https://avatars.githubusercontent.com/u/4200835?v=4
+tags: [rust, macros]
+description: In this post, we write a procedural macro that generates code to bind functions and types including arbitrary many impl blocks to a scripting language. The problems encountered and techniques learned can be applied to other tasks where complex compile-time analysis that spans multiple macro invocations is required.
+image: https://live.staticflickr.com/2785/4317096083_24697db960_b.jpg
+hide_table_of_contents: false
+---
+
+I recently wrote the most complex procedural Rust macro I’ve ever attempted. This post tries to outline the problems I’ve encountered and tells how I overcame them.
+
+## The Background
+
+With synth, we are building a declarative command line test data generator. For now, the specification that declares what test data to build is just JSON that gets deserialised to our data structures using `serde_json`. This was a quick and easy way to configure our application without much overhead in terms of code. For example:
+
+```json
```

Comment:
If you do ```json synth ...``` I think it will work with the playground.

Reply:
Yup - tested and it works :)

So you need to change the markdown code block language from `json` to `json synth`.

Reply:
I left out the `...`, but I think it should work nonetheless. At least that's what the other examples have, and it does look good locally.